sim_ws/src/arcus_gui/src/Global/Helper/QProcessHandler/QProcessHandler.cpp

```diff
@@ -0,0 +1,88 @@
+#include "QProcessHandler.hpp"
+
+#include <unistd.h>
+#include <sys/types.h>
+
+QProcessHandler::QProcessHandler(QWidget *parent_, std::string processName_, 
+    std::string bashCmd_, bool forwardPrint_) : QWidget(parent_),
+                                                _name(processName_),
+                                                _bashCmd(bashCmd_),
+                                                _forwardPrint(forwardPrint_)
+{
+    this->setupUi();
+    this->connectSignals();
+}
+
+void QProcessHandler::onPushed()
+{
+    if (!_processIsOn)
+    {
+        // Start the process by ensuring that the bash session is the leader of the process group with ros nodes
+        if (_forwardPrint)
+        {
+            _process.start("setsid", QStringList() << "bash" << "-c" << QString::fromStdString(_bashCmd));
+        }
+        else
+        {
+            _process.start("setsid", QStringList() << "xterm" << "-e" << "bash" << "-c" << QString::fromStdString(_bashCmd));
+        }
+        _processIsOn = true;
+    }
+    else
+    {
+        uint64_t pid = static_cast<uint64_t>(_process.processId());
+        if (pid > 0)
+        {
+            ::kill(-pid, SIGTERM);
+            _process.waitForFinished(PROCESS_KILL_TIMEOUT_MS);
+            RCLCPP_INFO(rclcpp::get_logger("GUI"), "Sent SIGTERM to process group with PID %ld", pid);
+
+            if (_process.state() != QProcess::NotRunning)
+            {
+                ::kill(-pid, SIGKILL);
+                RCLCPP_INFO(rclcpp::get_logger("GUI"), "Sent SIGKILL to process group with PID %ld", pid);
+                _process.waitForFinished(1000);
```

Comment:
and this PROCESS_KILL_TIMEOUT_MS
```suggestion
                _process.waitForFinished(PROCESS_KILL_TIMEOUT_MS);
```